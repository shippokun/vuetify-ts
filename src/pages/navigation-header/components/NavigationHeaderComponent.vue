<template>
  <v-app-bar app color="indigo" dark>
    <v-app-bar-nav-icon @click.stop="emitToggle()"></v-app-bar-nav-icon>
    <v-toolbar-title>Application</v-toolbar-title>
  </v-app-bar>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class NavigationHeaderComponent extends Vue {
  @Prop()
  public drawer!: boolean;

  @Prop()
  public toggleDraw!: Function;

  public emitToggle = () => {
    this.toggleDraw(!this.drawer);
  };
}
</script>
