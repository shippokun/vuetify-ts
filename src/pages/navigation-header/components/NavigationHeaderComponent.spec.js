import Vuetify from 'vuetify';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import NavigationHeaderComponent from './NavigationHeaderComponent.vue';

const localVue = createLocalVue();

describe('NavigationHeaderComponent.vue', () => {
  let vuetify;
  beforeEach(() => {
    vuetify = new Vuetify();
  });
  const wrapper = shallowMount(NavigationHeaderComponent, {
    localVue,
    propsData: {
      drawer: false,
    },
  });
  it('snapshots', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
