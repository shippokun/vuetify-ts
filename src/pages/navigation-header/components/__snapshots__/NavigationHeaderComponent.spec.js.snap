// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationHeaderComponent.vue snapshots 1`] = `
<v-app-bar-stub color="indigo" dark="true" tag="header" extensionheight="48" src="" value="true" app="true">
  <v-app-bar-nav-icon-stub></v-app-bar-nav-icon-stub>
  <v-toolbar-title-stub>Application</v-toolbar-title-stub>
</v-app-bar-stub>
`;
